### PR TITLE
CB-16737 Check aws parameters for null value and prevent NPE during e…

### PR DESCRIPTION
**Pull request by lturcsanyi**
…nv creation

See detailed description in the commit message.

### CODE REVIEW

**Review by lacikaaa — APPROVED** [review]

File: environment/src/test/java/com/sequenceiq/environment/environment/dto/EnvironmentDtoConverterTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    private EnvironmentDtoConverter underTest = new EnvironmentDtoConverter(
+            new HashMap<>(),
+            new HashMap<>(),
+            new AuthenticationDtoConverter(publicKeyValidator),
+            costTagging,
+            entitlementService,
+            defaultInternalAccountTagService,
+            new AccountTagToAccountTagResponsesConverter(),
+            accountTagService,
+            crnUserDetailsService);
```
Comment (lajosrodek) [inline environment/src/test/java/com/sequenceiq/environment/environment/dto/EnvironmentDtoConverterTest.java:46]: Not a big deal but I suggest initializing this in a `@BeforeEach setUp()` method. Ideally converters are indeed stateless, but it is still a best practice to instantiate the test victim anew for each test method.

File: environment/src/test/java/com/sequenceiq/environment/environment/dto/EnvironmentDtoConverterTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import com.sequenceiq.environment.environment.domain.Environment;
+import com.sequenceiq.environment.environment.domain.EnvironmentAuthentication;
+import com.sequenceiq.environment.environment.validation.validators.PublicKeyValidator;
+import com.sequenceiq.environment.tags.service.AccountTagService;
+import com.sequenceiq.environment.tags.service.DefaultInternalAccountTagService;
+import com.sequenceiq.environment.tags.v1.converter.AccountTagToAccountTagResponsesConverter;
+
+class EnvironmentDtoConverterTest {
+
+    private final PublicKeyValidator publicKeyValidator = Mockito.mock(PublicKeyValidator.class);
```
Comment (lajosrodek) [inline environment/src/test/java/com/sequenceiq/environment/environment/dto/EnvironmentDtoConverterTest.java:25]: Why not use `@Mock` on instance fields for automatic mock creation and `@ExtendWith(MockitoExtension.class)` on the test class?

**Review by lajosrodek — CHANGES_REQUESTED** [review]

File: environment/src/test/java/com/sequenceiq/environment/environment/dto/EnvironmentDtoConverterTest.java (hunk 2 above)
Comment (lturcsanyi) [reply environment/src/test/java/com/sequenceiq/environment/environment/dto/EnvironmentDtoConverterTest.java:25]: I prefer to use @Mock, but it seemed to me that in environment service this is the preferred way. I can change it if you like.

**Review by foldik — APPROVED** [review]
LGTM

**Review by lajosrodek — COMMENTED** [review]
Sorry, late to the party. LGTM :smile:

**Pull request merged**